cmd/create/testoperatorrelease/requests.go

```diff
@@ -0,0 +1,226 @@
+				}
+
+				if currentVersion.LessThan(minVersion) {
+					release.Spec.Components[i].Version = minVersion.String()
+				}
+				break
+			}
+		}
+
+		if found {
+			continue
+		}
+
+		for i, app := range release.Spec.Apps {
+			if app.Name == appname {
+				found = true
+				currentVersion, err := semver.NewVersion(app.Version)
+				if err != nil {
+					return microerror.Mask(err)
+				}
+
+				minVersion, err := semver.NewVersion(minVersionStr)
+				if err != nil {
+					return microerror.Mask(err)
+				}
+
+				if currentVersion.LessThan(minVersion) {
+					release.Spec.Apps[i].Version = minVersion.String()
+				}
+				break
+			}
+		}
+
+		if !found {
+			r.logger.LogCtx(ctx, "level", "warning", "message", fmt.Sprintf("Adding a new component from the requirements is not currently supported. Tried to add %s", appname))
+		}
+	}
+
+	return nil
+}
+
+func affects(r request, targetVersion semver.Version) (bool, error) {
+	// lemmas are expressions like '> 12.0.0, < 13.0.0'
+	lemmas := strings.Split(r.Name, ",")
+
+	for _, lemma := range lemmas {
+		lemma := strings.TrimSpace(lemma)
+
+		firstChar := lemma[0:1]
+		if firstChar == ">" || firstChar == "<" || firstChar == "=" {
+			// > 13.0.2, < 12.0.0, >= 13.0.0, <= 13.0.1, = 13.0.0
+			tokens := strings.Split(lemma, " ")
+			if len(tokens) != 2 {
+				return false, microerror.Maskf(invalidRequestError, "Unable to parse lemma %q", lemma)
+			}
+			operator := tokens[0]
+			// We assume the version in this case is without wildcards.
+			version, err := semver.NewVersion(tokens[1])
+			if err != nil {
+				return false, microerror.Mask(err)
+			}
+
+			switch operator {
+			case ">":
+				// Does not match if it is <=.
+				if targetVersion.LessThan(version) || targetVersion.Equal(version) {
+					return false, nil
+				}
+			case ">=":
+				// Does not match if it is <.
+				if targetVersion.LessThan(version) {
+					return false, nil
+				}
+			case "<":
+				// Does not match if it is >=.
+				if targetVersion.GreaterThan(version) || targetVersion.Equal(version) {
+					return false, nil
+				}
+			case "<=":
+				// Does not match if it is >.
+				if targetVersion.GreaterThan(version) {
+					return false, nil
+				}
+			case "=":
+				// Does not match if it is !=.
+				if !targetVersion.Equal(version) {
+					return false, nil
+				}
+			default:
+				return false, microerror.Maskf(invalidRequestError, "Unrecognized operator %q", operator)
+			}
+		} else {
+			// Wildcard things like "12.*"
+			reg := strings.ReplaceAll(lemma, ".", "\\.")
+			reg = strings.ReplaceAll(reg, "*", ".*")
+			re := regexp.MustCompile(reg)
+
+			if !re.Match([]byte(targetVersion.String())) {
+				return false, nil
+			}
+		}
+	}
+
+	return true, nil
+}
+
+// The mergeRequirements func takes a requests file and computes the list of applications and their minimum version.
+func (r *runner) mergeRequirements(ctx context.Context, reqs requests, targetVersion semver.Version) (map[string]string, error) {
```

Comment:
IIUC this function is constructing a map that contains the apps, and the **greatest** version for each of those apps specified in the requests that affect the target version. Can there be two different requests that affect the same target version requesting different app versions?

Reply:
there can very well be.
Imagine something like:
```
releases:
    - name: "> 12.0.0"
      requests:
        - name: node-exporter
          version: ">= 1.5.0"
    - name: "> 13.0.0"
      requests:
        - name: node-exporter
          version: ">= 1.7.0"
```